src/commands.ts

```diff
@@ -158,24 +159,67 @@ export class Commands {
       workspaceOwner = workspace.owner_name
       workspaceName = workspace.name
 
-      // TODO: multiple agent support
       const agents = workspace.latest_build.resources.reduce((acc, resource) => {
         return acc.concat(resource.agents || [])
       }, [] as WorkspaceAgent[])
 
       if (agents.length === 1) {
         folderPath = agents[0].expanded_directory
-      }
+        workspaceAgent = agents[0].name
+      }else{
+          const agentQuickPick = vscode.window.createQuickPick()
+          let lastAgents: WorkspaceAgent[]
+          agentQuickPick.title = `Connect to a agent`;
+  
+          agentQuickPick.busy = true;
+          lastAgents = agents
+          const agentItems: vscode.QuickPickItem[] = agents.map((agent) => {
+              let icon = "$(debug-start)"
+              if (agent.status !== "connected") {
+                icon = "$(debug-stop)"
+              }
+              return {
+                alwaysShow: true,
+                label: `${icon} ${agent.name}`,
+                detail: `${agent.name} • Status: ${agent.status}`,
+              }
+          })
+          agentQuickPick.items = agentItems
+          agentQuickPick.busy = false
+          agentQuickPick.show()
+  
+          const agent = await new Promise<WorkspaceAgent | undefined>((resolve) => {
+            agentQuickPick.onDidHide(() => {
+              resolve(undefined)
+            })
+            agentQuickPick.onDidChangeSelection((selected) => {
+              if (selected.length < 1) {
+                return resolve(undefined)
+              }
+              const agent = lastAgents[agentQuickPick.items.indexOf(selected[0])]
+              resolve(agent)
+            })
+          })
+          
+          if(agent != undefined){
+            folderPath = agent.expanded_directory
+            workspaceAgent = agent.name
+          }else{
+            folderPath = ''
+            workspaceAgent = ''
+          }
+        }
+      
     } else {
       workspaceOwner = args[0]
       workspaceName = args[1]
-      // workspaceAgent is reserved for args[2], but multiple agents aren't supported yet.
+      workspaceAgent = args[2]
       folderPath = args[3]
     }
 
     // A workspace can have multiple agents, but that's handled
     // when opening a workspace unless explicitly specified.
-    const remoteAuthority = `ssh-remote+${Remote.Prefix}${workspaceOwner}--${workspaceName}`
+    const remoteAuthority = `ssh-remote+${Remote.Prefix}${workspaceOwner}--${workspaceName}--${workspaceAgent}`
```

Comment:
This should be optionally appended, because when opening a stopped workspace there may be no agents.

Reply:
I've Changed this code